frontend/src/pages/petlistings/PetDetailCreate.jsx

```diff
@@ -0,0 +1,365 @@
+    const [selectedPhoto, setSelectedPhoto] = useState(0);
+    const [loading, setLoading] = useState(true);
+    const [err, setErr] = useState("");
+    const [photoUrls, setPhotoUrls] = useState([]);
+    const [photoObjects, setPhotoObjects] = useState([]);
+
+
+    const navigate = useNavigate();
+
+    const listingUrl = `${process.env.REACT_APP_API_URL}/api/petlistings`;
+
+    // upon page load: fetch petlisting and shelter
+    useEffect(() => {
+        setListingData(sampleListing);
+        const getReqContent = {
+            method: "GET",
+            headers: {
+                Authorization: `Bearer ${token}`,
+                'Content-Type': "application/json",
+            }
+        }
+
+        const fetchAllData = async () => {
+            try {
+                const listingOwner = await getShelterById(userId);
+                setShelterData(listingOwner);
+                // console.log(listingOwner);
+            } catch (err) {
+                setErr(err.message);
+                console.log("Uh oh =>", err);
+            }
+
+            setLoading(false);
+        }
+
+        const getShelterById = async (id) => {
+            if (role !== "shelter") {
+                throw new Error("you're not not a shelter")
+                return;
+            }
+
+            let shelterUrl = `${process.env.REACT_APP_API_URL}/api/shelters/${id}`;
+            const res = await fetch(shelterUrl, getReqContent);
+
+            if (res.status != 200) {
+                throw new Error(res.statusText);
+            }
+
+            const shelter = await res.json();
+            return shelter.data;
+        }
+
+        const getListingOwner = async (email) => {
+            const allShelters = [];
+            let hasNext = true;
+            let page = 1;
+            while (hasNext) {
+                let shelterUrl = `${process.env.REACT_APP_API_URL}/api/shelters?page=${page}`;
+                const res = await fetch(shelterUrl, getReqContent);
+
+                if (res.status != 200) {
+                    throw new Error(res.statusText);
+                }
+
+                const shelters = await res.json();
+                allShelters.push(...shelters.data);
+                hasNext = shelters.page.has_next;
+                page++;
+            }
+            const ownerShelter = allShelters.find(s => s.email === email);
+            return ownerShelter;
+        }
+
+        fetchAllData();
+        // end of useEffect()
+    }, []);
+
+
+    const processPhotos = (photos) => {
+        const defaultPhoto = "/images/logo_ref.png";
+        if (photos.length == 0) {
+            return [defaultPhoto];
+        }
+
+        if (Array.isArray(photos)) {
+            const photoUrls = photos.map((photo) => {
+                return `${process.env.REACT_APP_API_URL}/${photo.url}`;
+            });
+            return photoUrls;
+        }
+        return [defaultPhoto];
+    }
+
+    const stringToCode = (key) => {
```

Comment:
parser thingy

Reply:
If the user enters something like 'puppy' into the category due to them not knowing that dog / cat are what you're supposed to enter, then puppy will become 'Other'. So I'd say even with the parser having a placeholder hint would be nice.